File: ts/384/283.ts
// 双指针
function moveZeroes(nums: number[]): void {
  const len = nums.length
  let l = 0, r = 0
  while (r < len) {
    if (nums[r] !== 0) {
      if (nums[l] === 0) {
        let tmp = nums[r]
        nums[r] = nums[l]
        nums[l] = tmp
      }
      l += 1
    }
    r += 1
  }
};

File: c/384/283.c
void moveZeroes(int* nums, int numsSize){
  int l = 0, r = 0;
  while(r < numsSize) {
    if (nums[r] != 0) {
      if (nums[l] == 0) {
        int tmp = nums[r];
        nums[r] = nums[l];
        nums[l] = tmp;
      }
      l += 1;
    }
    r += 1;
  }
}